pyanaconda/network.py

```diff
@@ -1151,14 +1242,17 @@ def write_hostname(rootpath, ksdata, overwrite=False):
 
 def disableIPV6(rootpath):
     cfgfile = os.path.normpath(rootpath + ipv6ConfFile)
-    if ('noipv6' in flags.cmdline
-        and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
-                for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
-        log.info('Disabling ipv6 on target system')
-        with open(cfgfile, "a") as f:
-            f.write("# Anaconda disabling ipv6 (noipv6 option)\n")
-            f.write("net.ipv6.conf.all.disable_ipv6=1\n")
-            f.write("net.ipv6.conf.default.disable_ipv6=1\n")
+    try:
+        if ('noipv6' in flags.cmdline
+            and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
+                    for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
```

Comment:
the function in your proposed solution should be
```
    if _find_all_ethernet_devices_with_ipv6_not_ignored():
        return
```

Then why not go furhter with the function (it will never be reused in any case) and do just
```
if 'noipv6' in flags.cmdline \
    and _all_ethernet_devices_have_ipv6_ignored():
```
Which just says what the doc text above:
`"""Disable ipv6 if noipv6 boot option is set and all ethernet devices ignore ipv6"""`
which feels strange to me (although in general I prefer the code to tell what is happening rather than adding comments).